File: services/api/Models/SalonStatusLog.cs
namespace ClientFlow.Api.Models;

public class SalonStatusLog
{
    public Guid Id { get; set; }
    public Guid SalonId { get; set; }
    public AppUser? Salon { get; set; }
    public string FromStatus { get; set; } = string.Empty;
    public string ToStatus { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}

File: apps/admin/lib/models/salon_status_log.dart
class SalonStatusLog {
  SalonStatusLog({
    required this.id,
    required this.fromStatus,
    required this.toStatus,
    required this.createdAt,
  });

  final String id;
  final String fromStatus;
  final String toStatus;
  final DateTime createdAt;

  factory SalonStatusLog.fromJson(Map<String, dynamic> json) {
    return SalonStatusLog(
      id: json['id'] as String,
      fromStatus: json['fromStatus'] as String? ?? 'ACTIVE',
      toStatus: json['toStatus'] as String? ?? 'ACTIVE',
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
